Framework/DataHandling/inc/MantidDataHandling/LoadMuonStrategy.h

```diff
@@ -0,0 +1,50 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+#include "MantidAPI/Workspace_fwd.h"
+#include "MantidDataObjects/TableWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidGeometry/IDTypes.h"
+#include "MantidKernel/Logger.h"
+#include "MantidNexus/NexusClasses.h"
+
+namespace Mantid {
+namespace DataHandling {
+
+class DLLExport LoadMuonStrategy {
+public:
+  // Constructor
+  LoadMuonStrategy(Kernel::Logger &g_log, const std::string &filename);
```

Comment:
We need to be careful with references as parameters as the caller has to guarantee the lifetime of the object. For the loggers there is not much option but for things like strings and for class constructors then doing something like
```c++
LoadMuonStrategy(Kernel::Logger &g_log, std::string filename) : m_filename(std::move(filename)) {}
```
with `m_filename` just a `std::string` will mean the string is moved and only copied if necessary.